library/swift/src/Envoy.swift

```diff
@@ -1,7 +1,7 @@
 import Foundation
 
 @objcMembers
-public final class Envoy: NSObject {
+public final class Envoy<Engine: EnvoyEngine>: NSObject {
```

Comment:
I don't think this needs to be generic. @buildbreaker and I were considering just instantiating this class with an `EnvoyEngine.Type` (for class methods, or an instance of `EnvoyEngine`)

Reply:
It doesn't need to be I guess. Are there downsides?

I went this direction because it felt a little cleaner to me than passing the type as an initialization parameter that needs to be stored and referenced, esp. when the interface is static.

Reply:
Maybe it makes more sense in the context that our current thinking is that EnvoyEngine will always be static, and Envoy will be the platform instance.

Reply:
Generics are harder to pass around in platform code, because everything that uses them also needs to be generic. For example, we'd have to do:

```swift
func foo<T: EnvoyEngine>(_ envoy: Envoy<T>)
```

Instead of:

```swift
func foo(_ envoy: Envoy)
```

Reducing the generic and instead keeping the type hidden within the object will remove the need for this

Reply:
But nothing else should actually need the generic type right? Like `foo` doesn't need to know the specialized type of `Envoy`. Pretty sure `func foo(_ envoy: Envoy)` should still work(?)